webview/src/plots/components/comparisonTable/ComparisonTableRow.tsx

```diff
@@ -128,21 +134,47 @@ export const ComparisonTableRow: React.FC<ComparisonTableRowProps> = ({
             className={cx({ [styles.pinnedColumnCell]: pinnedColumn })}
             colSpan={pinnedColumn ? 1 : nbColumns}
           >
-            <div className={styles.rowPath}>
-              <button
-                className={styles.rowToggler}
-                onClick={toggleIsShownState}
-              >
-                <Icon icon={isShown ? ChevronDown : ChevronRight} />
-                <Tooltip
-                  content={path}
-                  placement="bottom-start"
-                  delay={NORMAL_TOOLTIP_DELAY}
+            <div className={styles.rowHeader} data-testid="row-header">
+              <div className={styles.rowHeaderPath}>
+                <button
+                  className={styles.rowToggler}
+                  onClick={toggleIsShownState}
                 >
-                  <span className={styles.pathText}>{path}</span>
-                </Tooltip>
-              </button>
-              <CopyButton value={path} className={styles.copyButton} />
+                  <Icon icon={isShown ? ChevronDown : ChevronRight} />
+                  <Tooltip
+                    content={path}
+                    placement="bottom-start"
+                    delay={NORMAL_TOOLTIP_DELAY}
+                  >
+                    <span className={styles.pathText}>{path}</span>
+                  </Tooltip>
+                </button>
+                <CopyButton value={path} className={styles.copyButton} />
+              </div>
+              {boundingBoxLabelsArr.length > 0 && (
+                <div className={styles.boundingBoxLabels}>
+                  <p className={styles.boundingBoxLabelsTitle}>Labels</p>
+                  {boundingBoxLabelsArr.map(([label, { color, selected }]) => (
+                    <React.Fragment key={label}>
+                      <input
+                        type="checkbox"
+                        id={color.slice(1)}
+                        name="labels"
+                        value={label}
+                        defaultChecked={selected}
+                        className={styles.hiddenInput}
+                      />
+                      <label
+                        className={styles.boundingBoxLabelsButton}
+                        style={{ background: color }}
+                        htmlFor={color.slice(1)}
+                      >
+                        {label}
+                      </label>
+                    </React.Fragment>
+                  ))}
+                </div>
+              )}
```

Comment:
Originally I was aiming for the labels to be on the far right like this:

![image](https://github.com/iterative/vscode-dvc/assets/43496356/9031d3d1-4018-4f61-885a-23464b23a5e1)

but I realized that the CSS to get that working is more complicated than I originally thought. As an alternative, I've moved them to the left and made both the path and classes sticky:

![Screenshot 2024-01-23 at 3 17 05 PM](https://github.com/iterative/vscode-dvc/assets/43496356/1d116f50-83a0-49a8-91b0-148faff392ad)

but there's probably a better way to make this design. Any other ideas?

Reply:
Only thing I would warn is that it's possible there could be many classes, so we might want to consider how to handle that.

Reply:
Thanks for the reminder! I was planning to update the labels to have some kind of  "Show More" button when there are too many. I'll add it to the todo list in this issue.

Reply:
This might be problematic when pinning columns 

<img width="616" alt="Screenshot 2024-01-24 at 10 31 46 AM" src="https://github.com/iterative/vscode-dvc/assets/3683420/b65dda77-280b-435f-8cb4-a3e2a6ea6e51">

Reply:
> This might be problematic when pinning columns

Good catch! I forgot to check on pinned columns. Hmmm, we need to either find a better place or just tweak the z-index 🤔 

![Screenshot 2024-01-24 at 10 07 21 AM](https://github.com/iterative/vscode-dvc/assets/43496356/40a0aece-4070-4a5f-8408-25f4ba443ae4)
![Screenshot 2024-01-24 at 9 56 26 AM](https://github.com/iterative/vscode-dvc/assets/43496356/4f0c2287-0882-4527-aabc-5c3a815d9ba4)

Reply:
I think the path shouldn't be longer than the first column.

One option would be to wrap the labels below the path and make sure it fits in the first column. The drawback would be that it would take more space.

Another option would be to use a dropdown multi-select menu. The biggest problem here is that always visible labels also act as a legend.

Could they be below the image instead? Like as another row? They could toggle with the image when clicking on the path and they'd have all of the path width more before wrapping.

Reply:
Thanks for the ideas! 

> I think the path shouldn't be longer than the first column.

Out of curiosity, why shouldn't the path be more than the first column? 

> Another option would be to use a dropdown multi-select menu. The biggest problem here is that always visible labels also act as a legend.

Agreed, I like keeping the labels visible for legend purposes. I am planning to put in a dropdown but it will function as a "Show More" button that will hold classes when there are too many. 

> Could they be below the image instead? Like as another row? They could toggle with the image when clicking on the path and they'd have all of the path width more before wrapping.

That could work! Example:

![Screenshot 2024-01-25 at 7 54 01 AM](https://github.com/iterative/vscode-dvc/assets/43496356/981faa19-9501-499f-b71e-cde79575d36e)

It does look a little weird when it comes to image by step plots. But maybe the styles just need to tweaked a bit. 

> One option would be to wrap the labels below the path and make sure it fits in the first column. The drawback would be that it would take more space.

Here's an example of placing the classes row above the images but separate from the path:

![Screenshot 2024-01-25 at 8 06 01 AM](https://github.com/iterative/vscode-dvc/assets/43496356/08c8f554-1867-43d5-a2f3-63fd9a78b8d1)

@sroy3, what do you think?

Reply:
> Out of curiosity, why shouldn't the path be more than the first column?

There is a line at the right of the pinned column and it would overflow otherwise

> Here's an example of placing the classes row above the images but separate from the path
I think that's what would work best. Do you think it would now be possible to align the classes right now? There'd be more room for them. If not, I think it's a great start until we figure out a better solution (if there is one).

Reply:
> I think that's what would work best. Do you think it would now be possible to align the classes right now? There'd be more room for them. If not, I think it's a great start until we figure out a better solution (if there is one).

Great! Yes, it shouldn't be difficult. I'll open a second chain PR that moves the classes below the path for easier reviewing and viewability.

Reply:
Updated design and fixed overflow issue in #5240